pokemon_battle_arena/src/arena/mod.rs

```diff
@@ -49,10 +52,12 @@ impl Arena {
     // Setter Methods
     //
     /// Sets the effect of the arena
-    pub fn set_effect(&mut self, new: enums::types) {
+    #[allow(dead_code)]
+    pub fn set_effect(&mut self, new: enums::Types) {

```

Comment:
Not sure if I agree with that. For POD, sure, but this isn't POD, so you might want to add checks to the setter later.